### ISSUE TITLE
question: customizing resolution algorithm

### ISSUE THREAD

**Issue by lucastschmidt** [angular-translate/angular-translate#1202]
I would like to be able to change the resolution algorithm to perform the following:
- 'modifier.page.key'
-- can be translated into *modifier.page.key* if found, if not, tries *page.key*, if not, tries *key*

Is there a built in way to customize the resolution algorithm, or do I need to replace an entire service or directive for that?

**Comment by tspaeth** [comment]
No, there is no built-in way for that as the "nested" resolution is deeply integrated in the library. You'll have to overwrite the functionality if you want to have a traversal on the dots regarding translation matching of each of them.

**Comment by lucastschmidt** [comment]
@tspaeth thanks for the answer. 

One more thing, do you know exactly which "component" needs to be replaced in this case? 

Anyways, thanks for the answers, closing this.

**Comment by tspaeth** [comment]
@lucastschmidt  Maybe a good point to jump in is "flatObject" here... https://github.com/angular-translate/angular-translate/blob/master/src/service/translate.js#L272

**Comment by lucastschmidt** [comment]
@tspaeth, I was able to change the resolution algorithm, to support the 'infiniteModifier.modifier.page.key'.

https://github.com/lucastschmidt/angular-translate/commit/225f4568db154c2a7978418149514cd188abda34

I dont know if it makes sense to add this feature to the lib. The way that I did, if the *translationId* exists in the object, it returns fine, otherwise it tries that crazy algorithm that I dont know the name, but for us makes a lot of sense. 

Basically, what we did is this:
- each page has its own *context* for translation, eg: *projects*.
- in this page we can then have a ok button, eg: *projects.ok*
- in another page we can have an ok button as well, eg *profile.ok*
- we can then create a translationId as *ok*
- if we wish to customize the translation in the *projects* page, we could do it, by translating *projects.ok*.

We can even support some more crazy customization, like some *custom dynamic modifier*. I did not test if a dynamic modifier would work, or how it could work in an angular app. But in theory, it may be possible for me to do something like: *github.projects.ok*, *anotherOrganization.projects.ok*. Not my requirement right now, so I did not test if there is a way to do something like that in angular.
